src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
-	{
-
-		FacadeRenderState facadeState = facades.get( side );
-		IBlockState blockState = facadeState.getSourceBlock();
-
-		builder.setDrawFaces( EnumSet.allOf( EnumFacing.class ) );
-
-		// Reset to no color multiplicator
-		builder.setColorRGB( 0xFFFFFF );
-		builder.useStandardUV();
-
-		// We only render the stilt if we don't intersect with any part directly, and if there's no part on our side
-		if( renderStilt && busBounds == null && layer == BlockRenderLayer.CUTOUT )
-		{
-			builder.setTexture( this.facadeTexture );
-			switch( side )
+			IBakedModel model = dispatcher.getModelForState( blockState );
+			try
 			{
-				case DOWN:
-					builder.addCube( 7, 1, 7, 9, 6, 9 );
-					break;
-				case UP:
-					builder.addCube( 7, 10, 7, 9, 15, 9 );
-					break;
-				case NORTH:
-					builder.addCube( 7, 7, 1, 9, 9, 6 );
-					break;
-				case SOUTH:
-					builder.addCube( 7, 7, 10, 9, 9, 15 );
-					break;
-				case WEST:
-					builder.addCube( 1, 7, 7, 6, 9, 9 );
-					break;
-				case EAST:
-					builder.addCube( 10, 7, 7, 15, 9, 9 );
-					break;
+				blockState = blockState.getBlock().getExtendedState( blockState, facadeAccess, pos );
 			}
-		}
-
-		// Do not add the translucent facade in any other layer than translucent
-		boolean translucent = AEApi.instance().partHelper().getCableRenderMode().transparentFacades;
-		if( translucent && layer != BlockRenderLayer.TRANSLUCENT )
-		{
-			return;
-		}
-
-		final float thickness = thinFacades ? 1 : 2;
-
-		BlockRendererDispatcher blockRendererDispatcher = Minecraft.getMinecraft().getBlockRendererDispatcher();
-		IBakedModel blockModel = blockRendererDispatcher.getModelForState( blockState );
-
-		final int color;
-		if( translucent )
-		{
-			color = 0x4CFFFFFF;
-		}
-		else
-		{
-			color = 0xFFFFFFFF;
-		}
-
-		// TODO: Cache this
-		for( EnumFacing facing : facadeState.getOpenFaces() )
-		{
-			TextureAtlasAndTint spriteAndTint = getSprite( blockModel, blockState, facing, rand );
-			if( spriteAndTint != null && spriteAndTint.sprite != null )
+			catch( Exception ignored )
 			{
-				// Use the tint color from the item stack here, which is based upon the assumption that the
-				// model used for the block will use the same meaning for tint indices as the item model does
-				if( spriteAndTint.tint != -1 )
-				{
-					int tintColor = facadeState.resolveTintColor( spriteAndTint.tint );
+			}
 
-					// Still apply the transparency color
-					tintColor &= 0xFFFFFF;
-					tintColor |= color & 0xFF000000;
-					builder.setColor( tintColor );
-				}
-				else
+			List<BakedQuad> modelQuads = new ArrayList<>();
+			//If we are forcing transparent facades, fake the render layer, and grab all quads.
+			if( transparent )
+			{
+				for( BlockRenderLayer forcedLayer : BlockRenderLayer.values() )
 				{
-					builder.setColor( color );
+					//Check if the block renders on the layer we want to force.
+					if( blockState.getBlock().canRenderInLayer( blockState, forcedLayer ) )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)